tests/foreman/endtoend/test_api_endtoend.py

```diff
@@ -1094,11 +1094,11 @@ def test_positive_end_to_end(self, function_sca_manifest, target_sat, rhel_conte
         # step 2.6: Enable a Red Hat repository
         rhel_repo = target_sat.api.Repository(
             id=target_sat.api_factory.enable_rhrepo_and_fetchid(
-                basearch='x86_64',
+                basearch=constants.DEFAULT_ARCHITECTURE,
                 org_id=org.id,
-                product=constants.PRDS['rhel'],
-                repo=constants.REPOS['rhst7']['name'],
-                reposet=constants.REPOSET['rhst7'],
+                product=constants.REPOS['rhsclient9']['product'],
```

Comment:
So you first set `rhel_ver_list` to `[settings.content_host.default_rhel_version]` and then here, you hardcode version to `9`? Doesn't make much sense to me. Maybe you can use `PRODUCT_KEY_SAT_CLIENT` + the actual version from settings?

Reply:
I agree with @lhellebr here. Let's not hardcode the version here.

Reply:
Good catch, updated as suggested